kv/dist_sender.go

```diff
@@ -266,7 +265,8 @@ func (ds *DistSender) sendRPC(desc *proto.RangeDescriptor, method string, args p
 	}
 
 	// Build a slice of replica addresses (if gossiped).
-	var addrs []net.Addr
+	// A node may have multiple stores, so the addr of store should include net addr and storeID
+	var addrs []rpc.StoreAddr
 	replicaMap := map[string]*proto.Replica{}
```

Comment:
Let's change replicaMap to be a map from StoreID => *proto.Replica. This will be a lot more efficient than generating the string representation of the inet address and concatenating store ID.

    replicaMap := map[int32]*proto.Replica{}

This further means that you will no longer need the StoreAddr.String() method and the getArg() method can just accept the StoreID, not the StoreAddr. This in turn will remove the need for the storeClient.getStoreAddr() method.